packages/lesswrong/components/ea-forum/EAEmojisHeader.tsx

```diff
@@ -0,0 +1,584 @@
+      void removeEmoji?.(userId);
+    }
+  }, [canRemove, userId, removeEmoji]);
+
+  const {LWTooltip, ForumIcon} = Components;
+  return (
+    <figure
+      className={classNames(classes.emoji, !emoji && classes.emojiCursor)}
+      style={{
+        left: `${x * window.innerWidth}px`,
+        top: `${y * EMOJIS_HEADER_HEIGHT}px`,
+      }}
+    >
+      <LWTooltip
+        title={description}
+        placement="bottom"
+        popperClassName={classes.goodNews}
+      >
+        <div style={{transform: `rotate(${theta}deg)`}}>
+          {emoji
+            ? (
+              link
+                ? (
+                  <Link
+                    to={link}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    className={classes.emojiLink}
+                  >
+                    {emoji}
+                  </Link>
+                )
+                : emoji
+            )
+            : <ForumIcon icon="AddEmoji" />
+          }
+        </div>
+      </LWTooltip>
+      {canRemove &&
+        <ForumIcon
+          icon="Close"
+          onClick={onRemove}
+          className={classes.removeButton}
+        />
+      }
+      {children}
+    </figure>
+  );
+}
+
+type Point = {x: number, y: number};
+
+const EmojiPlaceholder: FC<{hoverPos: Point}> = ({hoverPos}) => {
+  return (
+    <Emoji emoji={{
+      displayName: "",
+      userId: "",
+      theta: 0,
+      emoji: "",
+      link: "",
+      description: "",
+      ...hoverPos
+    }} />
+  );
+}
+
+const AddEmoji: FC<{insertPos: Point}> = ({insertPos}) => {
+  const {
+    setEmojis,
+    addEmoji,
+    insertEmoji,
+    setInsertEmoji,
+    onCancelInsert,
+    classes,
+  } = useEmojiContext();
+  const [anchorElement, setAnchorElement] = useState<HTMLElement | null>(null);
+  const [link, setLink] = useState("");
+  const [description, setDescription] = useState("");
+
+  const onInsert = useCallback(async () => {
+    const {x, y} = insertPos;
+    const theta = Math.round((Math.random() * MAX_THETA * 2) - MAX_THETA);
+    const result = await addEmoji(insertEmoji, link, description, x, y, theta);
+    const newEmojis = result.data?.AddBannerEmoji;
+    if (Array.isArray(newEmojis)) {
+      setEmojis(newEmojis);
+    }
+    onCancelInsert();
+  }, [insertPos, link, description, addEmoji, insertEmoji, setEmojis, onCancelInsert]);
+
+  const onClickAway = useCallback(() => {
+    if (!link && !description) {
+      onCancelInsert();
+    }
+  }, [onCancelInsert, link, description]);
+
+  const canAdd = !!(insertEmoji && link && description);
+
+  const {
+    LWPopper, LWTooltip, SectionTitle, EAButton, EAOnboardingInput,
+    LWClickAwayListener,
+  } = Components;
+  return (
+    <>
+      <LWPopper
+        open
+        clickable
+        placement="bottom-start"
+        anchorEl={anchorElement}
+        key={`${insertPos.x},${insertPos.y}`}
+      >
+        <LWClickAwayListener onClickAway={onClickAway}>
```

Comment:
I know there's some bug with `LWClickAwayListener` since the react 18 upgrade, but I tested it and it _seems_ to do what we want here.